akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowInspectSpec.scala

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (C) 2014-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.scaladsl
+
+import scala.util.control.NoStackTrace
+import akka.stream.ActorMaterializer
+import akka.stream.testkit._
+import akka.stream.testkit.Utils._
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class FlowInspectSpec extends StreamSpec {
+
+  implicit val materializer = ActorMaterializer()
+  import system.dispatcher
+
+  "A inspect" must {
+
+    "call the procedure for each element" in assertAllStagesStopped {
+      Source(1 to 3).inspect(testActor ! _).runWith(Sink.ignore) foreach {
```

Comment:
wireTap isn't too bad:
```
Source(1 to 3).wireTap(Sink.foreach(testActor ! _)).runWith(Sink.ignore)
```

just saying that it's value is on the edge, but ok with me